File: tlg/src/ch/bailu/tlg/StateLocked.java
package ch.bailu.tlg;

public class StateLocked extends State {
	private final static int ID = 4; 
	
	public StateLocked(InternalContext c) {
		super(c);
		context.setStatusText("Game over");
	}
	
	@Override
	public State init(PlatformContext c) {
		return this;
		
	}


	@Override
	public int getID() {
		return ID;
	}

}

File: tlg/src/main/kotlin/tlg/state/StateLocked.kt
package tlg.state

import tlg.context.InternalContext
import tlg.context.PlatformContext

class StateLocked(c: InternalContext) : State(c) {
    override fun init(c: PlatformContext): State {
        return this
    }

    override val id = ID

    override fun toString(): String {
        return "Game over"
    }

    companion object {
        private const val ID = 4
    }
}
